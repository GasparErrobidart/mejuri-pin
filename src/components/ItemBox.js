import React            from 'react'
import styled           from 'styled-components'
import { FiHeart }      from 'react-icons/fi';

import Button           from './Button'

const ItemBoxContainer = styled.div`
  background: ${({theme})=> theme.colors.white};
`

const Actions = styled.div`
  display: flex;
  padding: 20px 20px 20px 20px;
  flex-direction:column;
  align-items: center;
  & > *:last-child{
    margin-top:20px;
  }
`

const ItemBox = ({product, onLike, liked})=>{
  return (
    <ItemBoxContainer>
      <figure>
        <img src={product.variant_images[0].attachment_url_medium}/>
        <Actions>
          <figcaption>
            <h4>{product.name}</h4>
          </figcaption>
          <div>
            <Button className={liked ? "active" : ""} onClick={onLike}><FiHeart/> <span>{liked ? "Liked" : "Like"}</span></Button>
          </div>
        </Actions>
      </figure>


    </ItemBoxContainer>
  )
}

export default ItemBox
