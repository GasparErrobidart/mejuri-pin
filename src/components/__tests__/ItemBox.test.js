import '@testing-library/jest-dom'
import 'jest-styled-components'

import React            from 'react'
import { cleanup }      from '@testing-library/react'
import renderer         from 'react-test-renderer'
import {ThemeProvider}  from 'styled-components'

import ItemBox       from '../ItemBox'
import theme         from '../GlobalStyle/themes/mejuri-pin'
import testProduct   from '../../mock/product.js'

// MOCK GLOBAL EXTERNAL FUNCTION JIT_IMG (LAZY LOADING HANDLER)
global.JIT_IMG = ()=> true

// MOCK REACT REF FOR IMAGES
function createNodeMock(element) {
  if (element.type === 'img') {
    return {
      current : {},
    };
  }
  return null;
}

afterEach(cleanup)

test('Renders "ItemBox" component', () => {
  const options = {createNodeMock};
  const component       = renderer.create(
    <ThemeProvider theme={theme}>
      <ItemBox product={testProduct}/>
    </ThemeProvider>
  ,options);
  const tree            = component.toJSON()
  expect(tree).toMatchSnapshot();
});

/*
    TODO
    - Test getSrcSet method
    - Test start state
*/
