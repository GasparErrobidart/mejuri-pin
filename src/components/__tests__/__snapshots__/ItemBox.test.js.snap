// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders "ItemBox" component 1`] = `
.c2 {
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  font-size: 1rem;
  font-weight: 300;
  padding: 10px 10px;
  margin: 10px;
  color: #303030;
  box-shadow: 0px 2px 4px #aaa;
  text-transform: uppercase;
  background: transparent;
  cursor: pointer;
}

.c2:hover,
.c2:active,
.c2.active {
  box-shadow: 0px 0px 2px #aaa;
  color: #888888;
  background: #F3F3F3;
}

.c0 {
  background: #FFFFFF;
}

.c0.animating {
  -webkit-animation: pop-in 500ms;
  animation: pop-in 500ms;
}

.c0 img {
  width: 100%;
  opacity: 0;
  -webkit-transition: opacity 300ms ease-out;
  transition: opacity 300ms ease-out;
}

.c0 img.picture-perfect-ready {
  opacity: 1.0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 20px 20px 20px 20px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > *:last-child {
  margin-top: 20px;
}

@media (min-width:768px) {
  .c2 {
    padding: 10px 30px;
  }
}

<div
  className="c0 animating"
  onAnimationEnd={[Function]}
>
  <figure>
    <img
      alt="Spiral Earrings"
      lazy-srcset="
      https://dto508s2j2p46.cloudfront.net/system/spree/products/9869/mini/p52505557_Spiral_Earring_YG_both.png?1574627797 90w,
      https://dto508s2j2p46.cloudfront.net/system/spree/products/9869/small/p52505557_Spiral_Earring_YG_both.png?1574627797 230w,
      https://dto508s2j2p46.cloudfront.net/system/spree/products/9869/product/p52505557_Spiral_Earring_YG_both.png?1574627797 650w,
      https://dto508s2j2p46.cloudfront.net/system/spree/products/9869/large/p52505557_Spiral_Earring_YG_both.png?1574627797 1500w
    "
      sizes="1vw"
      src="/placeholder.png"
    />
    <div
      className="c1"
    >
      <figcaption>
        <h4>
          Spiral Earrings
        </h4>
      </figcaption>
      <div>
        <div
          className="c2"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20.84 4.61a5.5 5.5 0 0 0-7.78 0L12 5.67l-1.06-1.06a5.5 5.5 0 0 0-7.78 7.78l1.06 1.06L12 21.23l7.78-7.78 1.06-1.06a5.5 5.5 0 0 0 0-7.78z"
            />
          </svg>
           
          <span>
            Like
          </span>
        </div>
      </div>
    </div>
  </figure>
</div>
`;
